src/data/transform.ts

```diff
@@ -371,166 +326,133 @@ const bindLinearElementToElement = (
         });
         // to position the text correctly when coordinates not provided
         Object.assign(endBoundElement, {
-          y: end.y || excliadrawLinearElement.y - endBoundElement.height / 2,
+          y: end.y || linearElement.y - endBoundElement.height / 2,
         });
       } else {
-        endBoundElement = newElement({
-          x: endX,
-          y: endY,
-          width,
-          height,
-          ...existingElement,
-          ...end,
-          type: endType,
-        }) as ExcalidrawBindableElement;
+        switch (endType) {
+          case "rectangle":
+          case "ellipse":
+          case "diamond": {
+            endBoundElement = newElement({
+              x: endX,
+              y: endY,
+              width,
+              height,
+              ...existingElement,
+              ...end,
+              type: endType,
+            });
+            break;
+          }
+          default: {
+            assertNever(
+              linearElement as never,
+              `Unhandled element end type "${endType}"`,
+              true,
+            );
+          }
+        }
       }
 
       bindLinearElement(
-        excliadrawLinearElement,
+        linearElement,
         endBoundElement as ExcalidrawBindableElement,
         "end",
       );
     }
   }
   return {
-    linearElement: excliadrawLinearElement,
+    linearElement,
     startBoundElement,
     endBoundElement,
   };
 };
 
 class ElementStore {
-  res: ExcalidrawElement[] = [];
+  excalidrawElements = new Map<string, ExcalidrawElement>();
   elementMap = new Map<string, number>();
+  programmaticElementsWithId = new Map<string, ExcalidrawProgrammaticElement>();
```

Comment:
the `elementMap` and `programmaticElementsWithId` seem unused

Reply:
true but weird I didn't get any eslint warning for this :/